### PR TITLE
Migrating extensions from sarif-visualstudio-extension

### LINKED ISSUE #2157: Migrate suppressed and absent issue helpers from VS extension to SDK
Migrate the following helpers from the sarif-visualstudio-extension repo to the SDK repo:

- `Run.HasAbsentResults`
- `Run.HasSuppressedResults`
- `Result.IsSuppressed`

@michaelcfanning @eddynaka FYI

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #2157

### CODE REVIEW

File: src/Sarif/Extensions/ResultExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Extensions
+{
+    public static class ResultExtensions
```
Comment (lgolding) [inline src/Sarif/Extensions/ResultExtensions.cs:10]: >ResultExtensions [](start = 24, length = 16)

One we are in the SARIF SDK, these no longer need extension methods. The generated classes in `Sarif\Autogenerated` are `partial` classes, meaning that we can extend them with first-class methods. See the classes in `Sarif\Core`. That's where these methods now belong.

**Review by lgolding — CHANGES_REQUESTED** [review]
:clock1:

**Comment by lgolding** [comment]
Bring along the unit tests as well.

File: src/Sarif/Extensions/ResultExtensions.cs (hunk 1 above)
Comment (eddynaka) [reply src/Sarif/Extensions/ResultExtensions.cs]: fixed!

---
In reply to: [528762018](https://github.com/microsoft/sarif-sdk/pull/2159#discussion_r528762018) [](ancestors = 528762018)

**Comment by eddynaka** [comment]
done!

---
In reply to: [732213999](https://github.com/microsoft/sarif-sdk/pull/2159#issuecomment-732213999) [](ancestors = 732213999)

File: src/Sarif/Sarif.csproj [hunk 2]
```diff
@@ -32,7 +32,7 @@
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
     <PackageReference Include="System.Collections.Immutable" Version="5.0.0" />
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All"/>
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
```
Comment (lgolding) [inline src/Sarif/Sarif.csproj:35]: >GitHub [](start = 52, length = 6)

What is this for?

**Review by lgolding — APPROVED** [review]
:shipit:

File: src/Sarif/Core/Result.cs [hunk 3]
```diff
@@ -93,6 +93,23 @@ public ReportingDescriptor GetRule(Run run = null)
...
 
+        public bool IsSuppressed()
+        {
+            if (this == null)
+            {
+                throw new ArgumentNullException(nameof(Result));
+            }
+
+            IList<Suppression> suppressions = this.Suppressions;
+            if (suppressions == null || suppressions.Count == 0)
```
Comment (michaelcfanning) [inline src/Sarif/Core/Result.cs:104]: >suppressions == null [](start = 16, length = 20)

if suppressions is null, we don't have *any* suppressions related data in this log and so we can't answer the question. Two ideas, either throw an exception for this case of use bool TryIsSuppressed...

**Review by michaelcfanning — CHANGES_REQUESTED** [review]
:clock1:

File: src/Sarif/Core/Result.cs (hunk 3 above)
Comment (eddynaka) [reply src/Sarif/Core/Result.cs]: updated. One thing that i changed: we won't throw exception. What will happen is if the Result isn't available, then we will return false.

---
In reply to: [529072719](https://github.com/microsoft/sarif-sdk/pull/2159#discussion_r529072719) [](ancestors = 529072719)

Comment (michaelcfanning) [reply src/Sarif/Core/Result.cs]: that's right! in the TryXXX pattern, this is exactly the behavior you want. 

---
In reply to: [529483550](https://github.com/microsoft/sarif-sdk/pull/2159#discussion_r529483550) [](ancestors = 529483550,529072719)

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**